pandas/core/dtypes/missing.py

```diff
@@ -196,6 +197,8 @@ def _isna_ndarraylike(obj):
         else:
             values = obj
         result = values.isna()
+    elif isinstance(obj, ABCDatetimeArray):
```

Comment:
why is this not handled by the clause above?

Reply:
not until #24024